packages/openapi-v3/src/__tests__/unit/enhancers/spec-enhancer.extensions.unit.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {OASEnhancerService} from '../../..';
+import {SpecServiceApplication} from './fixtures/application';
+
+describe('spec-enhancer-extension-point', () => {
+  let app: SpecServiceApplication;
+  let specService: OASEnhancerService;
+
+  beforeEach(givenAppWithSpecComponent);
+  beforeEach(findSpecService);
+
+  it('setter - can set spec', async () => {
+    const EXPECTED_SPEC = {
+      openapi: '3.0.0',
+      info: {title: 'LoopBack Application', version: '1.0.0'},
+      // setter adds paths spec based on the default spec
+      paths: getSamplePathSpec(),
+    };
+    const PATHS_SPEC = getSamplePathSpec();
+    specService.spec = {...specService.spec, paths: PATHS_SPEC};
+    expect(specService.spec).to.eql(EXPECTED_SPEC);
+  });
+
+  it('generateSpec - loads and create spec for ALL registered extensions', async () => {
+    const EXPECTED_SPEC = {
+      openapi: '3.0.0',
+      // info object is updated by the info enhancer
+      info: {title: 'LoopBack Test Application', version: '1.0.1'},
+      paths: {},
+      // the security scheme entry is added by the security enhancer
+      components: {
+        securitySchemes: {
+          bearerAuth: {
+            type: 'http',
+            scheme: 'bearer',
+            bearerFormat: 'JWT',
+          },
+        },
+      },
+    };
+    const specFromService = await specService.generateSpec();
+    expect(specFromService).to.eql(EXPECTED_SPEC);
+  });
+
+  it('getEnhancerByName - returns the enhancer with a given name', async () => {
+    const enhancer = await specService.getEnhancerByName('info');
+    expect(enhancer).to.not.be.undefined();
+    expect(enhancer?.name).to.equal('info');
+  });
+
+  it('applyEnhancerByName - returns the merged spec after applying a given enhancer', async () => {
+    const EXPECTED_SPEC = {
+      openapi: '3.0.0',
+      // info object is updated by the info enhancer
+      info: {title: 'LoopBack Test Application', version: '1.0.1'},
+      paths: {},
+    };
+    const mergedSpec = await specService.applyEnhancerByName('info');
```

Comment:
Why `SECURITY_SCHEME_SPEC` is not part of the expected result? Does the enhancer only update the `info` part? Cause I thought (this line)[https://github.com/strongloop/loopback-next/pull/4258/files#diff-dd8c9f9bac61f1da3ef811d81198fb9aR40] merges the security spec to it.

Reply:
> Why `SECURITY_SCHEME_SPEC` is not part of the expected result? Does the enhancer only update the `info` part? Cause I thought (this line)[https://github.com/[/pull/4258](https://github.com/strongloop/loopback-next/pull/4258)/files#diff-dd8c9f9bac61f1da3ef811d81198fb9aR40] merges the security spec to it.

The service calls all contributors...including SecuritySpecEnhancer which applies the spec you are refering to.

Reply:
@agnes512 @emonddr By calling `applyEnhancerByName` you ONLY APPLY A SINGLE ENHANCER, in this case is `info` enhancer, that's why the security spec won't be included.

Let me rename the `generateSpec` function to sth like `applyAllEnhancers` to be more clear.